eth/domain.go

```diff
@@ -0,0 +1,34 @@
+//go:build !bignum_pure && !bignum_hol256
+// +build !bignum_pure,!bignum_hol256
+
+package eth
+
+import (
+	"math/big"
+
+	"github.com/protolambda/go-kzg/bls"
+)
+
+var (
+	BLSModulus *big.Int
+	DomainFr   []bls.Fr
+)
+
+func initDomain() {
+	BLSModulus = new(big.Int)
+	BLSModulus.SetString(bls.ModulusStr, 10)
+
+	// ROOT_OF_UNITY = pow(PRIMITIVE_ROOT, (MODULUS - 1) // WIDTH, MODULUS)
+	primitiveRoot := big.NewInt(7)
+	width := big.NewInt(int64(FieldElementsPerBlob))
+	exp := new(big.Int).Div(new(big.Int).Sub(BLSModulus, big.NewInt(1)), width)
+	rootOfUnity := new(big.Int).Exp(primitiveRoot, exp, BLSModulus)
+	DomainFr = make([]bls.Fr, FieldElementsPerBlob)
+	for i := 0; i < FieldElementsPerBlob; i++ {
+		// We reverse the bits of the index as specified in https://github.com/ethereum/consensus-specs/pull/3011
+		// This effectively permutes the order of the elements in Domain
+		reversedIndex := reverseBits(uint64(i), FieldElementsPerBlob)
+		domain := new(big.Int).Exp(rootOfUnity, big.NewInt(int64(reversedIndex)), BLSModulus)
```

Comment:
FWIW, I think we can speed this up by using a temporary variable for the output, multiplying it by the root of unity each iteration, and copying that into the output, and then shuffling the output to be in reverse bit order afterwards. But this works fine too.

Reply:
thx will keep in mind for the next iteration